File: app/src/main/java/avatar/com/mymvptestone/java/mvp/base/BaseMvp.java
package avatar.com.mymvptestone.java.mvp.base;

/**
 * $desc$
 *
 * @author Frank
 * email xiong.fu@avatarmind.com
 * created 19-4-4 下午1:39.
 */
public interface BaseMvp<M extends Model, V extends View, P extends BasePresenter> {
    /**
     * 创建model
     *
     * @return
     */
    M createModel();

    /**
     * 创建View
     *
     * @return
     */
    V createView();

    /**
     * 创建Presenter
     *
     * @return
     */
    P createPresenter();
}

File: app/src/main/java/avatar/com/mymvptestone/kotlin/mvp/base/BaseMvp.kt
package avatar.com.mymvptestone.kotlin.mvp.base

/**
 * $desc$
 * @author Frank
 * email xiong.fu@avatarmind.com
 * created 19-4-4 下午4:28.
 */
interface BaseMvp<M : Model, V : View, P : BasePresenter<M, V>> {
    fun createModel(): M
    fun createView(): V
    fun createPresenter(): P?
}